### PR TITLE
Send reminder

### PR DESCRIPTION
**Pull request by normanbos**
Both NPM `start` and `server` scripts must be executed.

### CODE REVIEW

File: src/components/ItemCard.js [hunk 1]
```diff
@@ -25,9 +30,59 @@ export default function ItemCard({
...
+      calculateTimeLeft()
+    }, 1000)
+  })
+
+  const [modalIsOpen, setIsOpen] = useState(false)
+  function openModal() {
+    setIsOpen(true)
+  }
+
+  function afterOpenModal() {}
```
Comment (jamarob) [inline src/components/ItemCard.js]: This does nothing and can be deleted

File: src/components/ItemCard.js [hunk 2]
```diff
@@ -25,9 +30,59 @@ export default function ItemCard({
...
   return (
     <CardWrapper>
+      <Modal
+        isOpen={modalIsOpen}
+        onAfterOpen={afterOpenModal}
+        onRequestClose={closeModal}
+        shouldCloseOnOverlayClick={false}
+        style={StyledModal}
+      >
+        <div>
```
Comment (jamarob) [inline src/components/ItemCard.js]: Is this div necessary?

File: src/components/closeModalCountdown.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React, { useEffect, useState } from 'react'
+import { ReminderButton } from './buttons'
+
+export default function CloseModalCountdown({ closeModal }) {
+  const [counter, setCounter] = useState(3)
```
Comment (jamarob) [inline src/components/closeModalCountdown.js]: This looks like it is foremost a custom hook, maybe extract it and use the hook in the parent component.

File: src/components/dueTimer.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import styled from 'styled-components'
+
+export default function DueTimer({ timeLeft }) {
```
Comment (jamarob) [inline src/components/dueTimer.js]: Maybe rename to `DueTime`, it doesn't really count time and just shows what is passed in.

File: src/components/itemCardDetails.js [hunk 5]
```diff
@@ -39,12 +39,11 @@ export default function ItemCardDetails({
         <CardHeader>
           <b>{title}</b>
           <CardHeaderNav>
-            <RoundButton onClick={handleEditToggle}>
-              <FaEdit />
-            </RoundButton>
-            <RoundButton onClick={() => deleteItem(item)}>
-              <FaTrashAlt />
-            </RoundButton>
+            {timeLeft.hours <= 0 && timeLeft.days <= 0 && (
```
Comment (jamarob) [inline src/components/itemCardDetails.js]: This check occurs a lot in your code, maybe extract it to a function?

File: src/components/reminderMailer.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    email: contact,
+    sent: false,
+    buttonText: 'Erinnerung senden',
+  })
+
+  function onSend(event) {
+    event.preventDefault()
+
+    setState({
+      name: borrower,
```
Comment (jamarob) [inline src/components/reminderMailer.js]: Spread in the state and only change the keys that have changed, like so 
```
setState({...state, sent: false, buttonText: '...wird gesendet'})
```

File: src/components/reminderMailer.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+    let data = {
+      name: state.name,
+      email: state.email,
+      message: state.message,
+    }
+
+    axios
+      .post('/api/v1', data)
+      .then(res => {
```
Comment (jamarob) [inline src/components/reminderMailer.js]: Maybe check the status code of res, you could receive a response that something has gone wrong on the server.

File: src/components/reminderMailer.js [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+          name: borrower,
+          message: `Hallo ${borrower}, bitte denk daran, mir folgenden Gegenstand wieder
+          zurück zu geben: ${item}`,
+          email: contact,
+          sent: true,
+          buttonText: 'Erinnerung gesendet',
+        })
+      })
+      .catch(() => {
+        console.log('Erinnerung nicht gesendet')
```
Comment (jamarob) [inline src/components/reminderMailer.js]: `buttonText` remains '...wird gesendet', maybe change it, to 'Nochmal versuchen' or something like that?

File: server.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+const { PORT = 4000 } = process.env
+
+const app = express()
+app.use(cors())
+app.use(express.json())
+
+app.use(express.static(path.join(__dirname, 'build')))
+
+app.get('/ping', function(req, res) {
+  return res.send('pong')
```
Comment (jamarob) [inline server.js]: Nice! :D

File: server.js [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+app.get('/', function(req, res) {
+  res.sendFile(path.join(__dirname, 'build', 'index.html'))
+})
+
+app.listen(PORT, () => console.log(`Server ready on http://localhost:${PORT}`))
+
+app.use(bodyParser.urlencoded({ extended: true }))
+
+app.get('/', (req, res) => {
```
Comment (jamarob) [inline server.js]: Is this route ever reached? I think the on line 18 will handle this request. Maybe you forgot to delete it?

File: server.js [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+app.post('/api/v1', (req, res) => {
+  var data = req.body
+
+  var smtpTransport = nodemailer.createTransport({
+    host: 'smtp.mailtrap.io',
+    port: 2525,
+    auth: {
+      user: '2dcd82b46f20b2',
+      pass: '71777e5b8854c6',
+    },
```
Comment (jamarob) [inline server.js]: Are those save to share on github?

File: cypress/fixtures/example.json [hunk 12]
```diff
@@ -0,0 +1,5 @@
+{
```
[the remark this answers is not in the document]
Comment (normanbos) [reply cypress/fixtures/example.json]: > 
> 
> I could not get your project to run.. :( please correct the following
> 
> `./src/App.js Cannot find file: 'buttons.js' does not match the corresponding name on disk: './src/components/Buttons.js'.`

Should work now.

File: src/components/itemCard.js [hunk 13]
```diff
@@ -25,9 +30,59 @@ export default function ItemCard({
...
+    setTimeLeft({
+      days: Math.floor(difference / (1000 * 60 * 60 * 24)),
+      hours: Math.floor((difference / (1000 * 60 * 60)) % 24),
+    })
+  }
+  useEffect(() => {
+    setTimeout(() => {
+      calculateTimeLeft()
+    }, 1000)
+  })
```
Comment (jamarob) [inline src/components/itemCard.js]: You forgot to provide a cleanup function, this produces a warning in the console, when the item is dismissed. Try this
```
useEffect(() => {
    const timer = setTimeout(() => {
      calculateTimeLeft()
    }, 1000)
    return () => clearTimeout(timer)
  })
```

File: src/components/ItemCard.js (hunk 2 above)
Comment (normanbos) [reply src/components/ItemCard.js]: Nope!

File: server.js (hunk 11 above)
Comment (normanbos) [reply server.js]: They are only for Mailtrap.

**Pull request merged**